tests/apitest/src/CoreServices/FSEventStream.cs

```diff
@@ -0,0 +1,169 @@
+		{
+			Console.WriteLine ("FsEventStream_WatchEvents");
+			watchedFileChanged = true;
+			//watchEventCalled.Set ();
+		}
+
+		[TearDown]
+		public void EndFSEventStream ()
+		{
+			fsEventStream.Stop ();
+			fsEventStream.Invalidate ();
+			fsEventStream.Dispose ();
+		}
+
+		[Test]
+		public void GetDeviceBeingWatchedTest ()
+		{
+			var deviceId = fsEventStream.GetDevice ();
+			Assert.AreEqual (st_dev, deviceId, "Device ID"); //Check if device id is the same
+		}
+
+		[Test]
+		public void FSEventFileChangedTest ()
+		{
+			string path = "FSEventStreamTests/TempFileToWatch.txt";
+			Assert.IsTrue (File.Exists (path));;
+
+			File.AppendAllText (path, "Hello World!");
+			//watchEventCalled.WaitOne (Int32.MaxValue);
+			//waitHandle.WaitOne ();
+			Assert.IsTrue (watchedFileChanged);
+		}
+
+		//[Test]
+		public void UnscheduleFromRunLoopTest ()
+		{
+			fsEventStream.UnscheduleFromRunLoop (CFRunLoop.Current);
+			File.AppendAllText ("FSEventStreamTests/TempFileToWatch.txt", "Hello World!");
+			watchEventCalled.WaitOne (Int32.MaxValue);
+			//waitHandle.WaitOne ();
+			//fsEventStream.Events
+			Assert.IsFalse (watchedFileChanged); //This shouldn't have changed as we unscheduled from loop?
+		}
+
+		//[Test]
+		public void SetExclusionPathsTest ()
+		{
+			using (var fsEventStreamForExclusion = new FSEventStream (st_dev, pathsToWatchRelativeToDevice, ulong.MaxValue, TimeSpan.MinValue, FSEventStreamCreateFlags.None)) {
+				fsEventStreamForExclusion.Events += FsEventStream_ExcludeWatchEvents;
+
+				using (var pathsToExclude = NSArray.FromStrings (new [] { "FSEventStreamTests" })) {
+					fsEventStreamForExclusion.SetExclusionPaths (pathsToExclude);
+					fsEventStreamForExclusion.ScheduleWithRunLoop (CFRunLoop.Current);
+					fsEventStreamForExclusion.Start ();
+
+					File.AppendAllText ("FSEventStreamTests/TempFileToWatch.txt", "Hello World!");
+					//watchEventCalled.WaitOne (Int32.MaxValue);
+					//excludeEventCalled.WaitOne (Int32.MaxValue);
+
+					Assert.IsTrue (watchedFileChanged); //fsEventStream callback should've changed this to true?
+					Assert.IsFalse (exludedFileChanged);  //fsEventStreamForExclusion callback shouldn't have changed this?
+				}
+
+			}
+		}
+
+		private void FsEventStream_ExcludeWatchEvents (object sender, FSEventStreamEventsArgs args)
+		{
+			Console.WriteLine ("FsEventStream_ExcludeWatchEvents");
+			exludedFileChanged = true;
+			excludeEventCalled.Set ();
+		}
+
+		//[Test]
+		public void SetDispatchQueueTest ()
+		{
+			using (var fsEventStreamDispatchQueue = new FSEventStream (st_dev, pathsToWatchRelativeToDevice, ulong.MaxValue, TimeSpan.MinValue, FSEventStreamCreateFlags.None)) {
+				fsEventStreamDispatchQueue.Events += FsEventStream_ExcludeWatchEvents;
+				fsEventStreamDispatchQueue.ScheduleWithRunLoop (CFRunLoop.Main);
+
+				fsEventStreamDispatchQueue.SetDispatchQueue (null); // shouldn't recieve events after this
```

Comment:
I'm not really sure how to test `SetDispatchQueue` apart from setting it to `null` to stop receiving events